### PR TITLE
snappy: add default security profile if none is specified

### PR DESCRIPTION
**Pull request by mvo5**
Instead of forcing people to explicitly add the migration-skill do so implicitly if no other skill is specified.

### CODE REVIEW

**Comment by zyga** [comment]
Looks good, +1

File: snappy/security.go [hunk 1]
```diff
@@ -71,6 +71,12 @@ var (
 		SecurityCaps: []string{},
 	}
 
+	// the default migration skill if no default is uses in
```
Comment (niemeyer) [inline snappy/security.go]: That doesn't sound right. We don't want the migration-skill to come in by default. We want people to explicitly mention it if they depend on anything non-default.

The permissions we discussed so far are:

- Writing into the applications own space. That should definitely come by default. It did so in the old convention, and it should continue to do so in the new convention.

- Network client access. That used to be a default in the old convention when nothing was specified, and we agreed to not have it by default in the new convention, at least for now. If people want network access, they need to explicitly mention migration-skill. That ensures we can go into either direction soon depending on what we decide at the sprint, without having to break anyone.

Is there anything else we're missing that exists in the default template of the old convention?

**Comment by mvo5** [comment]
@niemeyer Thanks for the review! I changed this now to the following:

- if there is a skill the security generation is done via the skill (currently only the migration skill)
- if there is no skill specified a default security policy is generated that has access to its own space and no security-caps, i.e. no network-client cap
- the migration-skill is left unchanged, the defaults of the old system are used *if* the user explicitly requests the migration skill (that means *if* the migration skill is used by default the network-client cap is granted) 

I hope this addresses the concerns. Please let me know if there is anything I can do.

File: snappy/security.go [hunk 2]
```diff
@@ -762,7 +768,12 @@ func generatePolicy(m *snapYaml, baseDir string) error {
 		if err != nil {
 			return err
 		}
+
+		// if no skill is specified, use the defaultSecurityPolicy
```
Comment (niemeyer) [inline snappy/security.go]: // TODO This is not actually right. Even if there are skills, we still want to give the snap
// a default set of allowances, such as being able to read and write in its own directory
// and perhaps network access (we're still deciding on that one). So the real logic we
// want here is: give the snap a default set of permissions, and then whatever else the
// skills permit (migration or not). This is coming soon.

**Comment by niemeyer** [comment]
LGTM, let's please just have the comment indicated replaced by that text so it's more clear where we are going.

**Comment by mvo5** [comment]
Thanks a bunch! Comment added.

**Comment by mvo5** [comment]
retest me please

**Comment by mvo5** [comment]
retest this please

**Comment by elopio** [comment]
retest this please

**Pull request merged**